### DISCUSSION TITLE
[Lesson6]Unable to enter private key

### DISCUSSION THREAD

**Discussion by Haruki65** [Cyfrin/foundry-full-course-cu#1243] [Q&A]
Even if I paste a correct private key in "Enter private key", it looks like this. (I know it doesn't appear)

Enter private key: 
Error: 
No arguments provided for contract constructor. Consider --constructor-args or --constructor-args-path

Context:
- Error #0: Failed to deploy contract
- Error #1: Invalid data

But, just as a test when I change a bit of the private key, it sometimes says "Odd number of digits" or "Invalid character '\u{1b}' at position 64". The private key from Ganache should be entered correctly. Please help me out with this..!

**Comment by xyz899** [comment]
Hi,

use for example on a script `forge script script/DeployFundMe.s.sol:DeployFundMe --rpc-url http://127.0.0.1:8545 --interactive` ( for privately copy pasting the private key) or `--private-key $PRIVATE_KEY` to paste it directly in the command line.

Be sure to verify that your private key has the correct length ans starts with 0x, if you are not sure use a notepad or a browser page so you can clearly see if the key is fully pasted or not and then just Ctrl+C , Ctrl+V enter and it should be good to go, if that ain't working use the .ENV style or makling an .env file and `source .env` or --private-key if its just a test key

**Reply by Haruki65** [reply]
Thank you for your clear solutions. Although I did --private-key, my terminal still doesn't work. 

forge create SSICerts --rpc-url http://172.26.240.1:7545 --private-key <the private key>
[⠢] Compiling...
No files changed, compilation skipped
Error: 
No arguments provided for contract constructor. Consider --constructor-args or --constructor-args-path

Context:
- Error #0: Failed to deploy contract
- Error #1: Invalid data

I double checked if the private key is correct but it's correct, and I can paste it to another platform. When I implement

forge script script/DeployFundMe.s.sol:DeployFundMe --rpc-url http://127.0.0.1:8545 --interactive

it gives this error

error: unexpected argument '--interactive' found

  tip: a similar argument exists: '--interactives'

I am new to foundry and programming, could you tell me more solutions or about source .env?

**Reply by xyz899** [reply]
Because your Contract is supposedly written with arguments that you didn't give in the command terminal, that's why its better to make a script generally for deploying a contract but like they said use :

`Consider --constructor-args or --constructor-args-path` 
 
 For the private key however, you can't use `--interactive` with `forge script`, you can use however `--interactives` or `--i` and set 1 as a specific account ( 0  being the default account ) for example here :

`forge script script/DeployAssets.s.sol:DeployAssets --rpc-url http://127.0.0.1:8545 -i 1 --sender <AddressAssociatedWithPrivateKey> --broadcast`

it will ask you then to enter a private key :

`Finding wallets for all the necessary addresses...
Enter private key:
`

##### Here I'm using the first anvil address one that starts with " 0xf39Fd6...266" and I put it as an argument for `--sender` and its private-key for `-i 1`
`
==========================

ONCHAIN EXECUTION COMPLETE & SUCCESSFUL.
Total Paid: 0.0054202 ETH (1355050 gas * avg 4 gwei)

`

**Reply by Haruki65** [reply]
Thank you again for your informative advice.

I tried some of the solutions you gave me, and I realized that I have a bad habit of not reading error messages precisely. 

When I implemented
forge create mycontract --rpc-url http://172.26.240.1:7545 --interactive --constructor-args "address"
[⠰] Compiling...
No files changed, compilation skipped
Enter private key: 
Deployer: ?
Deployed to: ?
Transaction hash: ?

It worked! I've spend hours to deal with the error, I am so relieved of the stress. Thank you anyway.
